elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/mapping/IndexMapping.java

```diff
@@ -0,0 +1,124 @@
+/*
+ *    Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License").
+ *    You may not use this file except in compliance with the License.
+ *    A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    or in the "license" file accompanying this file. This file is distributed
+ *    on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *    express or implied. See the License for the specific language governing
+ *    permissions and limitations under the License.
+ *
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.elasticsearch.mapping;
+
+import com.google.common.collect.ImmutableMap;
+import lombok.ToString;
+import org.elasticsearch.cluster.metadata.MappingMetaData;
+
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyMap;
+
+/**
+ * Elasticsearch index mapping. Because there is no specific behavior for different field types,
+ * string is used to represent field types.
+ */
+@ToString
+public class IndexMapping {
+
+    /**
+     * Field mappings from field name to field type in Elasticsearch date type system.
+     */
+    private final Map<String, String> fieldMappings;
+
+    public IndexMapping(Map<String, String> fieldMappings) {
+        this.fieldMappings = fieldMappings;
+    }
+
+    public IndexMapping(MappingMetaData metaData) {
+        this.fieldMappings = flatMappings(metaData.getSourceAsMap());
+    }
+
+    /**
+     * How many fields in the index (after flatten)
+     * @return  field size
+     */
+    public int size() {
+        return fieldMappings.size();
+    }
+
+    /**
+     * Return field type by its name.
+     * @param fieldName     field name
+     * @return              field type in string
+     */
+    public String getFieldType(String fieldName) {
```

Comment:
Is it possible to return null value? or we accept null value as return value?

Reply:
Yeah, it is. For now this is used in UT. Will add an existential check method or refactor it to return `Optional` if in use in future.